packages/jest-mock/src/index.ts

```diff
@@ -7,7 +7,7 @@
 
 /* eslint-disable local/ban-types-eventually, local/prefer-rest-params-eventually */
 
-export type MockFunctionMetadataType =
+export type MockMetadataType =
```

Comment:
Having `Function` in this and `MockFunctionMetadata` is misleading. Sounds like they are related with mock functions, but in fact these are definitions for mocked modules / objects. Might be helpful to change the names for future readers.